MarkBind/markbind
# Fix newline testcase

**What is the purpose of this pull request?**



Fixes #1726 

Copy of #1727, testing if this fixed the CI tests.

**Overview of changes:**
This PR fixed the issue raised in #1726.

This was caused by the extra newline created at the end of buildSiteNav() in the core/site dir. I simply added a trimEnd() to fix the issue.

ADDON: 
I have added the testing for the convert feature to catch the following cases: 
1. If the user already has a hierarchy of folders with different names 
2. If the user has a README file in the root directory
3. If the user does not have a stylesheet or content folder


**Anything you'd like to highlight / discuss:**

Since there are many file changes, to look for the files in question, the Home.html should not contain any unnecessary spaces in the navigation component


**Testing instructions:**

Run `markbind init -convert` on a test repo and see that there is no more new line before the closing </site-nav> tag.

**Proposed commit message: (wrap lines at 72 characters)**

Fixed `markbind convert` extra newline issue

---

**Checklist:** :ballot_box_with_check:
